lightningd/jsonrpc.c

```diff
@@ -576,7 +576,7 @@ static struct io_plan *jcon_connected(struct io_conn *conn,
 {
 	struct json_connection *jcon;
 
-	jcon = tal(ld, struct json_connection);
+	jcon = tal(conn, struct json_connection);
 	jcon->ld = ld;
 	jcon->used = 0;
 	jcon->buffer = tal_arr(jcon, char, 64);
```

Comment:
This means that finish_jcon should not do "tal_free(jcon)", as it has already been freed.  It passes because tal_free() actually calls destructors and marks everything as being-destroyed before it actually frees it, so the tal_free() there is a NOOP.

Reply:
In fact, jcon->log has already been freed too, and we use it in finish_jcon...